occurrences/admin.py

```diff
@@ -34,23 +34,51 @@ class OccurrenceInline(admin.TabularInline):
     model = Occurrence.study_groups.through
     extra = 0
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    # Without setting these add and delete permissions to false, the page loads
+    # slowly due to a large number of SQL queries
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
 
 class EventQueueEnrolmentInline(admin.TabularInline):
     model = EventQueueEnrolment
     extra = 0
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
+    # Without setting these add and delete permissions to false, the page loads
+    # slowly due to a large number of SQL queries
+    def has_add_permission(self, request, obj=None):
+        return False
+
+    def has_delete_permission(self, request, obj=None):
+        return False
+
 
 class EnrolmentInline(admin.TabularInline):
     model = Enrolment
     extra = 0
     exclude = ("person_deleted_at",)
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
 
 class StudyGroupInline(admin.TabularInline):
     model = StudyGroup
     extra = 0
     exclude = ("person_deleted_at",)
 
+    def has_change_permission(self, request, obj=None):
+        return False
+
```

Comment:
I think that here in `StudygroupInline` this `has_change_permission` is fine, but sure there can be cases that we may need to change the name of the group if it is misspelled or something. Since I don't think that fetching the permissions should be a big deal or a deal at all, I would'nt restrict editing.
```suggestion
```

Reply:
Sure. It's not necessary to restrict editing capability here either.